### PR TITLE
Refactoring and Completion of GetInfo functions of all statement and expression classes

### PR DESCRIPTION
**Pull request by haojin2**
This PR refactors the existing code for printing out debugging messages of all kinds of statements and adds additional implementations for previously missed statement types.
- Refactored GetInfo for SELECT, INSERT, DELETE, CREATE, UPDATE, and COPY statements.
- Added GetInfo for ANALYZE, PREPARE, DROP, EXECUTE, and TRANSACTION statements.
- Added GetInfo for aggregate, case, comparison, conjunction, constant value, function, operator, parameter value, and tuple value expressions.
- Re-formatted message of some types of statements and expressions.
- Corresponding tests
- Removed unnecessary legacy files in parser.

### CODE REVIEW

File: src/include/parser/create_statement.h [hunk 1]
```diff
@@ -225,6 +225,117 @@ class CreateStatement : public TableRefStatement {
 
   virtual void Accept(SqlNodeVisitor *v) override { v->Visit(this); }
 
+  const std::string GetInfo(int num_indent) const {
```
Comment (apavlo) [inline src/include/parser/create_statement.h]: I would this method private.

File: src/include/parser/delete_statement.h [hunk 2]
```diff
@@ -29,25 +29,38 @@ class DeleteStatement : public SQLStatement {
...
       table_ref->TryBindDatabaseName(default_database_name);
   }
 
-  std::string GetDatabaseName() const {
-    return table_ref->GetDatabaseName();
+  std::string GetDatabaseName() const { return table_ref->GetDatabaseName(); }
+
+  virtual void Accept(SqlNodeVisitor *v) override { v->Visit(this); }
+
+  const std::string GetInfo(int num_indent) const {
```
Comment (apavlo) [inline src/include/parser/delete_statement.h]: Again, make this private.

File: src/include/parser/delete_statement.h [hunk 3]
```diff
@@ -29,25 +29,38 @@ class DeleteStatement : public SQLStatement {
...
+    os << StringUtil::Indent(num_indent + 1) << GetTableName();
+    if (expr != nullptr) os << expr.get()->GetInfo() << std::endl;
+    return os.str();
   }
 
-  virtual void Accept(SqlNodeVisitor* v) override {
-    v->Visit(this);
+  const std::string GetInfo() const {
+    std::ostringstream os;
+
```
Comment (apavlo) [inline src/include/parser/delete_statement.h]: Unnecessary newline.

File: src/include/parser/transaction_statement.h [hunk 4]
```diff
@@ -33,8 +33,34 @@ class TransactionStatement : public SQLStatement {
...
-    v->Visit(this);
+  virtual void Accept(SqlNodeVisitor *v) override { v->Visit(this); }
+
+  const std::string GetInfo(int num_indent) const {
+    std::ostringstream os;
+    os << StringUtil::Indent(num_indent) << "TransactionStatement\n";
+    os << StringUtil::Indent(num_indent + 1) << "Type: ";
+    switch (type) {
+      case kBegin:
+        os << "Begin\n";
```
Comment (apavlo) [inline src/include/parser/transaction_statement.h]: add new line outside of the `switch` statement

File: test/parser/parser_utils_test.cpp [hunk 5]
```diff
@@ -123,34 +136,8 @@ TEST_F(ParserUtilTests, BasicTest) {
...
-                          (parser::CopyStatement*)stmt.get(), 0).size() > 0);
-          break;
-        case StatementType::UPDATE:
-          EXPECT_TRUE(parser::ParserUtils::GetUpdateStatementInfo(
-                          (parser::UpdateStatement*)stmt.get(), 0).size() > 0);
-          break;
-        default:
-          break;
-      }
+      LOG_INFO("%s\n", stmt->GetInfo().c_str());
```
Comment (apavlo) [inline test/parser/parser_utils_test.cpp]: I would make this `LOG_TRACE`

File: src/include/parser/create_statement.h (hunk 1 above)
Comment (haojin2) [reply src/include/parser/create_statement.h]: I plan to do this together with the phase of making all member variables private.

File: src/include/parser/delete_statement.h (hunk 3 above)
Comment (haojin2) [reply src/include/parser/delete_statement.h]: Done

File: src/include/parser/transaction_statement.h (hunk 4 above)
Comment (haojin2) [reply src/include/parser/transaction_statement.h]: Done

File: test/parser/parser_utils_test.cpp (hunk 5 above)
Comment (haojin2) [reply test/parser/parser_utils_test.cpp]: Done

**Comment by apavlo** [comment]
@haojin2 Is this ready for review?

**Comment by haojin2** [comment]
If you're okay with my putting the "making member variables in statements and expressions private" part in a separate PR then this is okay for review and merge.

File: src/include/expression/aggregate_expression.h [hunk 6]
```diff
@@ -116,10 +118,31 @@ class AggregateExpression : public AbstractExpression {
     }
   }
 
-  virtual void Accept(SqlNodeVisitor* v) override { v->Visit(this); }
+  virtual void Accept(SqlNodeVisitor *v) override { v->Visit(this); }
+
+  const std::string GetInfo(int num_indent) const override {
```
Comment (apavlo) [inline src/include/expression/aggregate_expression.h]: I thought that we were moving these into the .cpp files? And I thought that we were making the `GetInfo(int num_indent)` methods protected?

File: src/parser/create_statement.cpp [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+const std::string CreateStatement::GetInfo(int num_indent) const {
+  std::ostringstream os;
+  os << StringUtil::Indent(num_indent) << "CreateStatement\n";
+  os << StringUtil::Indent(num_indent + 1);
+
+  switch (type) {
+    case CreateStatement::CreateType::kTable: {
+      os << "Create type: Table"
+         << "\n";
```
Comment (apavlo) [inline src/parser/create_statement.cpp]: Use `std::endl`

Comment (apavlo) [inline src/parser/create_statement.cpp]: Also, move all of these newlines at the end of a `case` statement to be outside the `switch`

File: src/parser/execute_statement.cpp [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+// table_ref.cpp
+//
+// Identification: src/parser/execute_statement.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <sstream>
+#include "parser/execute_statement.h"
```
Comment (apavlo) [inline src/parser/execute_statement.cpp]: Put the corresponding header for this .cpp file at the top of the list of includes.

File: src/parser/create_statement.cpp (hunk 7 above)
Comment (haojin2) [reply src/parser/create_statement.cpp]: Done

Comment (haojin2) [reply src/parser/create_statement.cpp]: Done

File: src/include/expression/aggregate_expression.h (hunk 6 above)
Comment (haojin2) [reply src/include/expression/aggregate_expression.h]: I thought the change from all public to protected or private will be the next phase?

File: src/parser/execute_statement.cpp (hunk 8 above)
Comment (haojin2) [reply src/parser/execute_statement.cpp]: Done

**Comment by haojin2** [comment]
@apavlo All code review addressed, should be good for merge after builds pass.

**Comment by apavlo** [comment]
@haojin2 Can you resolve the new conflicts?

**Comment by haojin2** [comment]
Done with resolving conflicts, should be good for merge once builds pass.

**Comment by haojin2** [comment]
Three checks failed with the LLVM error, the other two passed, I think this should be good to go.

**Pull request merged**